Cooper.Controllers/GameController.cs

```diff
@@ -37,24 +37,31 @@ public IActionResult GetAll()
         }
 
         /// <summary>
-        /// Returns game by id.
+        /// Returns game by name.
         /// </summary>
         /// <returns>Game</returns>
         /// <response code="200">If the game exists</response>
         /// <response code="404">If the game does not exist</response>
         [HttpGet]
-        [Route("{id}")]
+        [Route("{name}")]
         [ProducesResponseType(typeof(Game), StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
-        public IActionResult GetGameById(long id)
+        public IActionResult Get(string name)
         {
-            Game game = gameRepository.Get(id);
-
+            Game game = null;
+            if (long.TryParse(name, out long id))
+            {
+                game = gameRepository.Get(id);
+            }
+            else
+            {
+                game = gameRepository.Get(name);

```

Comment:
is Name property in DB uniq?

Reply:
Yes

Reply:
I don't like... What if we have a game with name that will be parsable to long?
"2048", "10e2", "666", any other strange name) can we split searching by name and by id?